prow/config/inrepoconfig.go

```diff
@@ -108,13 +108,25 @@ func prowYAMLGetter(
 
 	mergeMethod := c.Tide.MergeMethod(orgRepo)
 	log.Debugf("Using merge strategy %q.", mergeMethod)
+	if err := ensureHeadCommits(repo, headSHAs...); err != nil {
+		return nil, fmt.Errorf("failed to fetch headSHAs: %v", err)
+	}
 	if err := repo.MergeAndCheckout(baseSHA, string(mergeMethod), headSHAs...); err != nil {
 		return nil, fmt.Errorf("failed to merge: %w", err)
 	}
 
 	return ReadProwYAML(log, repo.Directory(), false)
 }
 
+func ensureHeadCommits(repo git.RepoClient, headSHAs ...string) error {
+	for _, sha := range headSHAs {
+		if err := repo.Fetch("origin", sha); err != nil {
+			return err
```

Comment:
I would do `fmt.Errorf("failed to fetch headSHA: %s", sha)` here and return err above so that it's easier to debug

Reply:
Done